File: server/src/commands/default/about.py
from .. import register_command

import socket
import yaml

from src.vars import chat_name, short_ver, codename, server_edition, authors, ext_ver
from src.colors import *
from yaml import SafeLoader
from init import server_dir, User, ClientSender

# Open Configuration
with open(server_dir + "/config.yml") as config_data:
        config = yaml.load(config_data, Loader=SafeLoader)

@register_command("about")
def about_command(socket: socket.socket, user: User, args: list, sender: ClientSender):
    sender.send(f"""{GREEN +  Colors.UNDERLINE + Colors.BOLD}About {chat_name}{RESET + Colors.RESET}
        {BLUE + Colors.BOLD}Thank you for using {chat_name}!{RESET}
        {BLUE + Colors.BOLD}Version: {RESET}{short_ver} {codename} ({server_edition}) ({ext_ver})
        {BLUE + Colors.BOLD}Author: {RESET}{", ".join(authors)}{RESET + Colors.RESET}""")

File: src/commands/default/about.rs
use libstrawberry::colors::{BLUE, BOLD, C_RESET, CYAN, GREEN, RESET, UNDERLINE};

use crate::system_core::commands;
use crate::system_core::commands::CommandCategory;
use crate::system_core::internals::MessageToClient;
use crate::system_core::permissions::Permissions;
use crate::global::{AUTHORS, CHAT_NAME, CODENAME, CORE_VERSION, DEFAULT_VERSION, EXT_VERSION, SERVER_EDITION};

pub fn about() -> commands::Command {
    async fn logic(ctx: &commands::Context) -> commands::CommandResponse {
        let message = format!("
    {GREEN}{BOLD}{UNDERLINE}About {CHAT_NAME}{C_RESET}
  * {CYAN}{BOLD}Thank you for using {CHAT_NAME}!{C_RESET}
  * {BLUE}{BOLD}Version:{RESET} {} {CODENAME} ({SERVER_EDITION}) ({}){RESET}{C_RESET}
  * {BLUE}{BOLD}Core Version:{RESET} {}{RESET}{C_RESET}
  * {BLUE}{BOLD}Author:{RESET} {} {RESET}{C_RESET}

  - {BLUE}{CYAN}https://github.com/Strawberry-Foundations/strawberry-chat{C_RESET}",
        *DEFAULT_VERSION, *EXT_VERSION, *CORE_VERSION, AUTHORS.join(", "));


        ctx.tx_channel.send(MessageToClient::SystemMessage {
            content: message
        }).await.unwrap();

        Ok(None)
    }

    commands::Command {
        name: "about".to_string(),
        aliases: vec![],
        description: "Shows about description for Strawberry Chat".to_string(),
        category: CommandCategory::Default,
        permissions: Permissions::Member,
        required_args: 0,
        handler: |ctx| Box::pin(async move {
            logic(&ctx).await
        }),
    }
}